core/src/main/java/org/apache/gravitino/cache/CaffeineEntityCache.java

```diff
@@ -330,6 +349,32 @@ private <KEY, VALUE> Caffeine<KEY, VALUE> newBaseBuilder(Config cacheConfig) {
     return (Caffeine<KEY, VALUE>) builder;
   }
 
+  private NameIdentifier getNameIdentifier(Entity entity) {
+    NameIdentifier nameIdent =
+        NameIdentifier.of(((HasIdentifier) entity).namespace(), ((HasIdentifier) entity).name());
+    return nameIdent;
+  }
+
+  private EntityCacheKey makeEntityCacheKey(Entity entity) {
+    NameIdentifier nameIdent = getNameIdentifier(entity);
+    EntityCacheKey entityCacheKey = EntityCacheKey.of(nameIdent, entity.type());
+    return entityCacheKey;
+  }
+
+  public void putReverseIndex(Entity entity, EntityCacheRelationKey key) {
```

Comment:
All entities in cache has the type:  `<E extends Entity & HasIdentifier>`, not `Entity`.

So it should be like this:

```
  public <E extends Entity & HasIdentifier> void putReverseIndex(E entity, EntityCacheRelationKey key) {
```